NGitLab.Mock/Clients/IssueClient.cs

```diff
@@ -106,7 +106,15 @@ public IEnumerable<Models.Issue> ForProject(int projectId)
         {
             using (Context.BeginOperationScope())
             {
-                return GetProject(projectId, ProjectPermission.View).Issues.Select(i => i.ToClientIssue()).ToList();
+                var project = GetProject(projectId, ProjectPermission.View);
+
+                return project
+                    .Issues
+                    .Where(i => project.CanUserViewConfidentialIssues(Context.User)
+                        || i.Author.Id == Context.User.Id
+                        || i.Assignee.Id == Context.User.Id)
```

Comment:
I don't think this is true, https://docs.gitlab.com/ee/user/project/issues/confidential_issues.html#permissions-and-access-to-confidential-issues

> However, a guest user can also create confidential issues, but can only view the ones that they created themselves.

Reply:
I removed the free pass for assignees.